### PR TITLE
Enhance Testing Resilliancy

### PR DESCRIPTION
**Pull request by adamoutler**
A few unit tests were intermittently failing during long dev cycles/repeated testing within the devcontainer.  This PR enhances reliability and repeatability. 

## Summary by CodeRabbit

* **Tests**
  * Enhanced Docker Compose test cleanup operations with improved orphan container and volume pruning.
  * Improved log collection with retry mechanisms for better handling of timing edge cases.
  * Strengthened test diagnostics and post-run visibility.
  * Refactored test infrastructure for improved command tracking and subprocess simulation accuracy.

### CODE REVIEW

File: test/docker_tests/test_docker_compose_scenarios.py [hunk 1]
```diff
@@ -484,6 +501,39 @@ def _run_with_conflict_retry(run_cmd: list[str], run_timeout: int) -> subprocess
...
+                best_result = logs_result
+
+            # If we see the expected startup marker, logs are complete
+            if "Startup pre-checks" in combined or "NETALERTX_CHECK_ONLY" in combined:
+                break
+
+            if attempt < max_attempts - 1:
+                time.sleep(wait_between)
+
+        return best_result or logs_result
```
[the remark this answers is not in the document]
Comment (adamoutler) [reply test/docker_tests/test_docker_compose_scenarios.py:537]: fixed
